### PR TITLE
feat: created school and faculty controller/model.

### LINKED ISSUE #19: Super Admin should be able to create School
Super Admin on the system should be able to create a School on the system with guidance from the School Schema in https://dbdiagram.io/d/600ec8ee80d742080a37bebd. A user should be assigned to the system as a School Admin

### LINKED ISSUE #20: Faculty creation
As a School Admin, I  want to be able to create Faculties in my School using the model specified in https://dbdiagram.io/d/600ec8ee80d742080a37bebd.

### PR DESCRIPTION
**Pull request by bynalab**

### CODE REVIEW

File: app/Http/Controllers/FacultyController.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        
+
+        $checkSchool = Faculty::where(['name' => $facultyName])->first();
+        if(empty($checkSchool)){
+            if($faculty->save()){
+                return response()->json(['success' => true], 200);
+            }
+            return response()->json(['success' => false], 200);
+        }
+        return response()->json(['error' => 'This faculty already exist'], 200);
```
Comment (Auwalms) [inline app/Http/Controllers/FacultyController.php]: @bynalab is there a reason we are returning status code 200 for an error response? Looks like a `409` will be ideal for this case, see https://stackoverflow.com/questions/3825990/http-response-code-for-post-when-resource-already-exists for details

File: app/Http/Controllers/SchoolController.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        $school->status = $status;
+
+        $checkSchool = School::where(['name' => $schoolName])->first();
+        if(empty($checkSchool)){
+            if($school->save()){
+                return response()->json(['success' => true], 200);
+            }
+            return response()->json(['success' => false], 200);
+        }
+        return response()->json(['error' => 'This school already exist'], 200);
```
Comment (Auwalms) [inline app/Http/Controllers/SchoolController.php]: Same as previous comment, this response code should be 409 not 200. see https://stackoverflow.com/questions/3825990/http-response-code-for-post-when-resource-already-exists

File: app/Http/Controllers/SchoolController.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create(Request $request)
+    {
+        $id = $this->uuid();
+        $schoolName = $request->get('schoolName');
+        $schoolCode = $request->get('schoolCode');
+        $status = 'Active';
```
Comment (Auwalms) [inline app/Http/Controllers/SchoolController.php]: We will need this status thingy to be  an enum, Active or Inactive. But yes default should be active

File: database/migrations/2021_02_14_150913_create_faculties_table.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('faculties', function (Blueprint $table) {
+            $table->string('id')->primary();
+            $table->string('school_id');
+            $table->string('name');
+            $table->string('code');
+            $table->enum('status', ['Active', 'Inactive']);
```
Comment (Auwalms) [inline database/migrations/2021_02_14_150913_create_faculties_table.php:21]: Great.

File: app/Http/Controllers/SchoolController.php [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create(Request $request)
+    {
+        $id = $this->uuid();
+        $schoolName = $request->get('schoolName');
```
Comment (Auwalms) [inline app/Http/Controllers/SchoolController.php:35]: Biko where are we getting the school name and school code from? I no sabi Laravel, mind explaining?

**Review by Auwalms — CHANGES_REQUESTED** [review]
Please check through my comments.

File: app/Http/Controllers/SchoolController.php (hunk 2 above)
Comment (bynalab) [reply app/Http/Controllers/SchoolController.php]: Noted

File: app/Http/Controllers/SchoolController.php (hunk 3 above)
Comment (bynalab) [reply app/Http/Controllers/SchoolController.php]: It's an enum, you'll see from the db migration. Front end will send the value

File: app/Http/Controllers/SchoolController.php (hunk 5 above)
Comment (bynalab) [reply app/Http/Controllers/SchoolController.php:35]: From the frontend. Post request

File: app/Http/Controllers/SchoolController.php (hunk 3 above)
Comment (Auwalms) [reply app/Http/Controllers/SchoolController.php]: Seen.

File: app/Http/Controllers/SchoolController.php (hunk 5 above)
Comment (Auwalms) [reply app/Http/Controllers/SchoolController.php:35]: Ahh  PHP ehnn, I will pass sha

File: app/Http/Controllers/SchoolController.php (hunk 2 above)
Comment (Auwalms) [reply app/Http/Controllers/SchoolController.php]: Can you correct this and push so we merge the PR? Thank you.

**Review by Auwalms — APPROVED** [review]
Looks good.

File: app/Http/Controllers/SchoolController.php (hunk 2 above)
Comment (Auwalms) [reply app/Http/Controllers/SchoolController.php]: Thanks.

File: app/Http/Controllers/FacultyController.php (hunk 1 above)
Comment (Auwalms) [reply app/Http/Controllers/FacultyController.php]: Thanks for the update.

**Review by abdullahikawu — APPROVED** [review]
Looks good

**Pull request merged**